apis/kueue/v1beta1/resourceflavor_types.go

```diff
@@ -78,6 +79,11 @@ type ResourceFlavorSpec struct {
 	// +optional
 	// +listType=atomic
 	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:validation:XValidation:rule="self.all(x, !has(x.key) ? x.operator == 'Exists' : true)", message="operator must be Exists when 'key' is empty, which means 'match all values and all keys'"
```

Comment:
Just for discussion, from maintenance, do you guys think this is more simple comparing to the codes.

Reply:
From maintenance POV it depends on how the rules look, From a runtime POV if we are able to fully  drop the webhook for a type (like it was the case for local queue) then it's a win.

Reply:
Make sense, apply the same policy through the whole project is important to avoid confusion, unless we can't fully remove the validation.